src/librustc_typeck/check/demand.rs

```diff
@@ -539,6 +541,41 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     return Some((sp, "consider removing the borrow", code));
                 }
             }
+            (
+                _,
+                &ty::RawPtr(TypeAndMut { ty: _, mutbl: hir::Mutability::Not }),
+                &ty::Ref(_, _, hir::Mutability::Not),
```

Comment:
While I believe we could also do something for mutable references (both for staying in the same mutability and moving to const), this requires some care to get right without false positives, so maybe just leave a FIXME or open an issue?

Reply:
Yes, I missed the mutable references situation. I'll open another issue.